### PR TITLE
Feature/include bundle discount on active

### PR DESCRIPTION
**Pull request by FredrikAreschoug**

### CODE REVIEW

File: src/main/java/com/hedvig/contract/service/PremiumServiceImpl.kt [hunk 1]
```diff
@@ -28,6 +32,15 @@ class PremiumServiceImpl(
     }
   }
 
+  override fun calculateTotalPremiumCostsOfPeriod(memberId: String, period: YearMonth): MonetaryAmount {
+    val agreementPremiumCosts = calculatePremiumCostsOfPeriod(memberId, period)
+    return if (agreementPremiumCosts.isEmpty()) {
+      Money.of(BigDecimal.ZERO, "SEK")
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/service/PremiumServiceImpl.kt:38]: 🤔 is it going to be sad with only _**SEK**_

File: src/main/java/com/hedvig/contract/shared/models/ContractState.kt [hunk 2]
```diff
@@ -101,6 +103,25 @@ abstract class ContractState<TAgreement : AgreementState>(
       }
     }
 
+  val currentOrAgreementStartingThisMonth: TAgreement?
+    get() {
+      val today = LocalDate.now(DateUtil.STOCKHOLM_ZONE_ID)
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/shared/models/ContractState.kt:108]: We can get the timezone from the contract, just to be 100% legit. No need to change that though

File: src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt [hunk 3]
```diff
@@ -86,13 +99,20 @@ class CampaignServiceImpl(
...
+  }
+
+  private fun getBundleDiscountOrNull(typesToBeBundled: List<InsuranceType>, currency: CurrencyUnit): Campaign? = if (
+    typesToBeBundled.size == 2
+    && typesToBeBundled.any { it == InsuranceType.NORWEGIAN_YOUTH_HOME_CONTENTS }
+    && typesToBeBundled.any { it == InsuranceType.NORWEGIAN_YOUTH_TRAVEL }
     && currency.currencyCode == "NOK"
   ) {
-    Money.of(BigDecimal(49), currency)
+    Campaign(_code = "NorwegianYouthBundle", incentive = Incentive.MonthlyCostDeduction(MonetaryAmountV2(BigDecimal(49).toString(), currency.currencyCode)))
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt:115]: Should we add the actual amount into a config?

File: src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt [hunk 4]
```diff
@@ -175,13 +209,17 @@ class CampaignServiceImpl(
...
+      val premiumTotalCosts = premiumService.calculateTotalPremiumCostsOfPeriod(memberId, YearMonth.now())
+      currentTotalPrice = premiumTotalCosts.asBigDecimal()
+      preferredCurrency = premiumTotalCosts.currency.currencyCode
 
     } else {
+      /*
+        TODO: we need to figure out how not use /latestQuote, or `getActiveDiscounts` should not be used used when
+        there is no contracts let's try to remove this at some point and redirect them to `getInsuranceCostOnPrice` or
+        `getBundledInsuranceCostOnPrice`
+      */
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt:222]: 🤔

Comment (meletios) [reply src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt:222]: is it comments for the future? Should we log something if we end up there?

File: src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt (hunk 3 above)
Comment (FredrikAreschoug) [reply src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt:115]: yeah why not

File: src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt (hunk 4 above)
Comment (FredrikAreschoug) [reply src/main/java/com/hedvig/productPricing/campaign/service/CampaignServiceImpl.kt:222]: Yes we could log the stack trace to see if it used

**Comment by FredrikAreschoug** [comment]
Refactored some in f908513 so new reviews are needed :)

**Pull request merged**